src/main/java/com/example/hw5/adapter/CountAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.hw5.adapter
+
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import coil.load
+import coil.size.Scale
+import coil.size.ViewSizeResolver
+import com.example.hw5.model.GithubUser
+import com.example.hw5.databinding.ItemUserBinding
+
+class CountAdapter(
+
+    private val itemClick: (GithubUser) -> Unit
+) : ListAdapter<GithubUser, UserViewHolder>(DIFF_UTIL) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        return UserViewHolder(
+            binding = ItemUserBinding.inflate(layoutInflater, parent, false),
+            onUserClicked = itemClick
+        )
+    }
+
+    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
+        holder.bind(getItem(position))
+
+    }
+
+    companion object {
+        private val DIFF_UTIL = object : DiffUtil.ItemCallback<GithubUser>() {
+            override fun areItemsTheSame(
+                oldItem: GithubUser,
+                newItem: GithubUser
+            ): Boolean {
+                return oldItem.id == newItem.id
+            }
+
+            override fun areContentsTheSame(
+                oldItem: GithubUser,
+                newItem: GithubUser
+            ): Boolean {
+                return oldItem.login == newItem.login

```

Comment:
ты сравниваешь элементы только по login, тут лучше либо полное сравнение, либо по тем элементам которые используешь
```suggestion
                return oldItem.login == newItem.login && oldItem.avatarUrl == newItem.avatarUrl
```
если у тебя у пользователя поменяется картинка ты элемент не обновишь